code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -757,32 +775,22 @@ butcher
 	var/mob/living/critter/flock/drone/F = holder.owner
 	if(!F?.flock)
 		return
-	for(var/atom/T in view(target_range, holder.owner))
-		//handle vehicles first
-		if (isvehicle(T))
-			var/vehicle_is_enemy = F.flock.isEnemy(T) //if someone gets in an enemy pod, attack
-			var/enemy_found = FALSE
-			for (var/mob/occupant in T)
-				if (!F.flock.isEnemy(occupant) && !vehicle_is_enemy)
-					continue
-				F.flock.updateEnemy(occupant)
-				enemy_found = TRUE
-				break
-			if (enemy_found) //bad guy in pod
-				. += T
-			//continue regardless of whether we find an enemy or not, since we don't want to attack empty pods
-			continue
-		if(!F.flock.isEnemy(T))
+
+	for(var/atom/T in F.flock.enemies)
+		if(istype(T.loc, /obj/flock_structure/cage)) //already got the bastard, don't need to do anything else
 			continue
-		if(isliving(T))
-			var/mob/living/M = T
-			if(is_incapacitated(M))
-				continue
-		// mob is a valid target, check if they're not already in a cage
-		if(!istype(T.loc.type, /obj/flock_structure/cage))
-			// if we can get a valid path to the target, include it for consideration
+		if (isvehicle(T.loc))
+			if(T.loc in view(holder.owner, target_range))
+				F.flock.updateEnemy(T)
+				F.flock.updateEnemy(T.loc)
+				. += T.loc
+		else if(T in view(holder.owner,target_range))
+			F.flock.updateEnemy(T)
+			if(isliving(T))
+				var/mob/living/M = T
+				if(is_incapacitated(M))
+					continue
```

Comment:
Why doesn't this call `isEnemy` anywhere?

Reply:
Because we're iterating over the list of enemies